lib/Controller/BanController.php

```diff
@@ -55,6 +58,17 @@ public function banActor(string $actorType, string $actorId, string $internalNot
 			$moderatorActorType = $moderator->getActorType();
 			$moderatorActorId = $moderator->getActorId();
 
+			$bannedParticipant = $this->participantService->getParticipantByActor($this->room, $actorType, $actorId);
+			
+			if ($bannedParticipant->getAttendee()->getParticipantType() === Participant::MODERATOR) {
+				throw new \InvalidArgumentException('Cannot ban a moderator');
+			}
+
+			if ($actorId === $moderatorActorId && $actorType === $moderatorActorType) {
+				throw new \InvalidArgumentException('Cannot ban yourself');
```

Comment:
Just as a note, if you check the error message in the integration test, you will notice this never triggers, as you yourself are a moderator so the previous check does it :P

Reply:
Yeah I also thought the same but yesterday I was able to see that it was able catch this check. so I just left it But your right today it never triggers so I just moved it up.